promising/utils.py

```
@@ -77,3 +49,36 @@ def get_running_asyncio_loop(*, raise_if_none: bool = True) -> AbstractEventLoop
+
+def assert_no_sync_usage_deadlock(loop_of_future: AbstractEventLoop, message: str) -> None:
+    try:
+        running_loop = asyncio.get_running_loop()
+    except RuntimeError:
+        running_loop = None
+
+    if running_loop is loop_of_future:
+        raise SyncUsageError(message)
+
+
+async def awaitable_as_coroutine(awaitable: Awaitable[Any]) -> Any:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `awaitable_as_coroutine`

(ANN401)